wasm/src/logic/mod.rs

```diff
@@ -45,6 +45,41 @@ impl LogicContext {
             self.handle_message(msg)?;
         }
 
+        // TODO: Remove this temporary sprite loading. Replace it with some kind of
+        // "resource complete" event
+        {
+            use rask_engine::resources::GetStore;
+            let res = crate::communication::RESOURCE_TABLE.read();
+            let tex1: Result<&rask_engine::resources::Texture, _> = res.get(0);
+            let tex2: Result<&rask_engine::resources::Texture, _> = res.get(1);
+            let texes = (if tex1.is_ok() { 1 } else { 0 }) + (if tex2.is_ok() { 1 } else { 0 });
+            if self.state.sprites().len() < texes {
+                let (res, id) = match (tex1, tex2) {
+                    (Ok(tex1), Err(_)) => (tex1, 0),
+                    (Ok(tex1), Ok(tex2)) => {
+                        if self.state.sprites().is_empty() {
+                            (tex1, 0)
+                        } else {
```

Comment:
is only one texture uploaded per tick?

Reply:
Yes, but this is not relevant, because this is very temporary code only for demonstrational purposes